apps/gdalalg_vector_geom_make_valid.cpp

```diff
@@ -0,0 +1,144 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  "gdal vector geom make-valid"
+ * Author:   Even Rouault <even dot rouault at spatialys.com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, Even Rouault <even dot rouault at spatialys.com>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#include "gdalalg_vector_geom_make_valid.h"
+
+#include "gdal_priv.h"
+#include "ogrsf_frmts.h"
+
+//! @cond Doxygen_Suppress
+
+#ifndef _
+#define _(x) (x)
+#endif
+
+/************************************************************************/
+/*                 GDALVectorGeomMakeValidAlgorithm()                   */
+/************************************************************************/
+
+GDALVectorGeomMakeValidAlgorithm::GDALVectorGeomMakeValidAlgorithm(
+    bool standaloneStep)
+    : GDALVectorGeomAbstractAlgorithm(NAME, DESCRIPTION, HELP_URL,
+                                      standaloneStep, m_opts)
+{
+    AddArg("keep-lower-dim", 0,
```

Comment:
Other options are available (could be added later)

https://github.com/libgeos/geos/blob/1e4d0b01fe307b347fb629793baf8488146d041b/capi/geos_c.h.in#L3221

Reply:
I've added a commit that adds a --method=linework|structure argument